lib/fs.js

```diff
@@ -3028,6 +3029,36 @@ function mkdtempSync(prefix, options) {
   return binding.mkdtemp(prefix, options.encoding);
 }
 
+/**
+ * Synchronously creates a unique temporary directory.
+ * The returned value is a disposable object which removes the
+ * directory and its contents when disposed.
+ * @param {string | Buffer | URL} prefix
+ * @param {string | { encoding?: string; }} [options]
+ * @returns {object} A disposable object with a "path" property.
+ */
+function mkdtempDisposableSync(prefix, options) {
+  options = getOptions(options);
+
+  prefix = getValidatedPath(prefix, 'prefix');
+  warnOnNonPortableTemplate(prefix);
+
+  const path = binding.mkdtemp(prefix, options.encoding);
+  // we need to stash the full path in case of process.chdir()
+  const fullPath = pathModule.resolve(process.cwd(), path);
+
+  const remove = () => {
+    binding.rmSync(fullPath, 0 /* maxRetries */, true /* recursive */, 100 /* retryDelay */);
+  };
+  return {
+    get path() {
+      return path;
+    },
+    remove,
```

Comment:
If you're just going to return an anonymous object, may as well just make both `path` and `remove` regular own properties. It likely doesn't make much sense to have one be a getter and the other be a regular property.